Extension conflicts (#8475)
fishtown-analytics/docs.getdbt.com: website/docs/docs/about-dbt-extension.md

- **Working with YAML files:** Today, the dbt extension has the following limitations with operating on YAML files:
  - Go-to-definition is not supported for nodes defined in YAML files (like snapshots).
  - Renaming models and columns will not update references in YAML files.
  - Future releases of the dbt extension will address these limitations.

- **Renaming models:** When you rename a model file, the dbt extension applies edits to update all `ref()` calls that reference the renamed model. Due to limitations of VS Code's Language Server Client, the extension can't auto-save these edited files. As a result, renaming a model file may cause compiler errors in your project. To fix these errors, either manually save each file that the dbt extension edited, or click **File** --> **Save All** to save all edited files.

- **Using Cursor's Agent mode:** When using the dbt extension in Cursor, lineage visualization works best in Editor mode and doesn't render in Agent mode. If you're working in Agent mode and need to view lineage, switch to Editor mode to access the full lineage tab functionality.

> ### Known conflicts with other extensions

Reviewer: maybe we can shorten this since its a bit long and already under the 'known limitations' section
```suggestion
### Extension conflicts
```

The extension may occasionally conflict with other VS Code extensions that provide similar services (such as code validation). You may need to disable these third-party extensions while working with the dbt extension.

> #### YAML by Red Hat

Author: ```suggestion
**YAML by Red Hat:**
```

> The extension may occasionally conflict with other VS Code extensions that provide similar services (such as code validation). You may need to disable these third-party extensions while working with the dbt extension. The following are known conflicts:

Reviewer: ```suggestion
The extension may occasionally conflict with other VS Code extensions that provide similar services (such as code validation). You may need to disable these third-party extensions while working with the dbt extension.
```

> #### YAML by Red Hat

Reviewer: are we having more conflicts added soon? having this as an h4 makes it seem like there will be future expansion and reads like its a scaffolding for a list that doesn't exist yet. since its only one conflict so far, i'd recommend downgrading this to a bolded text
```suggestion
The most common example today is:

**YAML by Red Hat**
```

Author: That's a question mark. We don't know what we don't know so far, not opposed to turning this bold. We can revert later if more are added, but I would avoid "common" just because i don't know how common it is

> The YAML extension by Red Hat may cause some keys (such as `static_analysis`) in dbt YAML files to be erroneously flagged as invalid in the IDE. 

Reviewer: sugg active voice
```suggestion
The YAML extension by Red Hat may erroneously flag some keys (such as `static_analysis`) in dbt YAML files as invalid in the IDE.
```

<Lightbox src="/img/docs/extension/false-yaml-error.png" width="60%" title="Static analysis erroneously tagged as invalid"/>

> To solve this issue, do one of the following:

Reviewer: love this clearness!
